packages/core/index.js

```diff
@@ -96,17 +96,19 @@ const runRequest = async (
     request.error = e
     try {
       await runMiddlewares(request, onErrorMiddlewares, plugin)
-      // Catch if onError stack hasn't handled the error
-      if (request.response === undefined) throw request.error
     } catch (e) {
       // Save error that wasn't handled
       e.originalError = request.error
       request.error = e
+
       throw request.error
     }
+    // Catch if onError stack hasn't handled the error
+    if (request.error && request.response === undefined) throw request.error
```

Comment:
`request.error && ` is redundant

Reply:
True, the middleware may have changed the error to throw.
So what about `throw request.error || e`

Reply:
I don't expect someone to override the Error to something falsy or nullish (famous last words). You can leave the `|| e` out for now.

Reply:
Fixed